app/models/vm_or_template/operations/configuration.rb

```diff
@@ -74,7 +74,7 @@ def disconnect_floppies
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
     if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
+      datastore = ext_management_system.storages.find_by(:name => options[:datastore])
```

Comment:
@masayag You would need to access hosts from the `ext_management_system` and collect the `writable_storages`.

Maybe something like this:
```ruby
storages = ext_management_system.hosts.collect(&:writable_storages).uniq.compact
datastore = storages.detect { |storage| storage.name == options[:datastore] }
```